ballerina/types.bal

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public type ConnectionConfig record {|
+    string baseUrl;
+    int identityMapCapacity;
+    map<json> originals;
+    map<string> headers;

```

Comment:
Best practice to add separate record types for the configuration fields. Any reason not to follow that ?

Reply:
Here, since the fields are not deterministic, they do not have a fixed structure for fields. 
And also key fields in the `originals` should be in the format `{string}.{string}.{string}` for Java APIs and mapping some undeterministic record fields to this format can be complex.
And the decision to use`json` as the value type in the `originals` map, as it aligns with the Java type `Map<String, Object>` from the native code. And there are some known fields which require `string[]`, `int`, `string` values which suggests to have a flexible data type for this map

Reply:
Shall we use following structure then ?
```suggestion
    map<anydata> originals;
    map<string> headers;
```